src/main/java/com/clevercloud/biscuitpulsar/AuthorizationProviderBiscuit.java

```diff
@@ -210,173 +147,61 @@ public void initialize(ServiceConfiguration conf, PulsarResources pulsarResource
+        );
+        Set<String> rules = Set.of("right($tenant, $namespace, $operation) <- namespace($tenant, $namespace), namespace_operation($operation), " + namespaceOperations + ".contains($operation)");
+        Set<String> checks = Set.of("check if right(" + namespaceFragment(namespaceName) + "," + namespaceOperationFragment(operation) + ")");
+        return authorize(() -> defaultProvider.allowNamespaceOperationAsync(namespaceName, role, operation, authData), true, "allowNamespaceOperationAsync(" + operation + " -> " + namespaceName + ")", role, authData, facts, rules, checks);
     }
 
     @Override
     public CompletableFuture<Boolean> allowNamespacePolicyOperationAsync(NamespaceName namespaceName, PolicyName policy, PolicyOperation operation, String role, AuthenticationDataSource authData) {
-        if (!role.startsWith("biscuit:")) {
-            return defaultProvider.allowNamespacePolicyOperationAsync(namespaceName, policy, operation, role, authData);
-        }
-
-        Either<Error, Verifier> res = verifierFromBiscuit(role);
-        if (res.isLeft()) {
-            return CompletableFuture.failedFuture(new Exception(res.getLeft().toString()));
-        }
-
-        Verifier verifier = res.get();
-        verifier.set_time();
-
-        verifier.add_fact(namespaceFact(namespaceName)).get();
-        verifier.add_fact(namespacePolicyOperationFact(namespaceName, policy, operation)).get();
-        verifier.add_rule("right(#authority, $tenant, $namespace, $operation) <- right(#authority, #admin), namespace_operation(#ambient, $tenant, $namespace, $operation), " + policiesOperations + ".contains($operation)").get();
-        verifier.add_check("check if right(#authority," + namespaceFactFragment(namespaceName) + "," + policyOperationFact(policy, operation) + ")").get();
-        verifier.allow();
-
-        Either verifierResult = verifier.verify(runLimits);
-        log.debug(verifier.print_world());
-
-        if (verifierResult.isLeft()) {
-            log.debug("Biscuit allowNamespacePolicyOperationAsync [{}]:[{}] on [{}] NOT authorized for role [{}]: {}", policy, operation, namespaceName, role, verifierResult.getLeft());
-            return isSuperUser(role, authData, this.conf);
-        } else {
-            log.debug("Biscuit allowNamespacePolicyOperationAsync [{}]:[{}] on [{}] authorized.", policy, operation, namespaceName);
-            return CompletableFuture.completedFuture(true);
-        }
+        Set<String> facts = Set.of(
+                namespaceFact(namespaceName),
+                namespacePolicyOperationFact(policy, operation)
+        );
+        Set<String> rules = Set.of("right($tenant, $namespace, $operation) <- namespace($tenant, $namespace), namespace_operation($operation), " + policiesOperations + ".contains($operation)");
+        Set<String> checks = Set.of("check if right(" + namespaceFragment(namespaceName) + "," + namespacePolicyOperationFragment(policy, operation) + ")");
```

Comment:
are we sre that no injection is possible here with the namespace name?

Reply:
https://github.com/apache/pulsar/blob/4f1e39b6921ea401b8c27f17a041d06d85f8abf8/pulsar-common/src/main/java/org/apache/pulsar/common/naming/NamespaceName.java#L181

The provided `NamespaceName` from pulsar is checked by pulsar on its creation.